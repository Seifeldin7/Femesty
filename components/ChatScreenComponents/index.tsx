export * from './ChatInputBox'
export * from './ChatMessage'