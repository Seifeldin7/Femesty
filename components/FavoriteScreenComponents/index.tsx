export * from "./FavoriteHzScroller";
export * from "./FavoriteLook";
