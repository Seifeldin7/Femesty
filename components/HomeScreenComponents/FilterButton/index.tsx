import React from "react";
import { View, Image } from "react-native";
import styled from "styled-components/native";

import { styles } from "./styles";

const Menu = styled.TouchableOpacity`
  align-items: center;
`;

const MenuText = styled.Text`
  font-size: 9px;
  color: "rgba(255,255,255,1)";
`;

export const FilterButton = () => {
  return (
    <View style={styles.container}>
      <Menu>
        <Image
          source={require("../../../assets/icons/small_icons/filter.png")}
          style={styles.image}
        />
        <MenuText active="true">Filter</MenuText>
      </Menu>
    </View>
  );
};
