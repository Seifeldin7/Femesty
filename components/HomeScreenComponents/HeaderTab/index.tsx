import React from "react";
import styled from "styled-components/native";
import { View } from "react-native";

import { styles } from "./styles";

const Menu = styled.Text`
  color: #fff;
  letter-spacing: 0.8px;
  font-weight: ${(props) => (props.bold ? "bold" : "normal")};
  opacity: ${(props) => (props.bold ? 1 : 0.8)};
  font-size: ${(props) => (props.bold ? "16px" : "15px")};
`;
const Separator = styled.View`
  width: 1px;
  height: 13px;
  background-color: #d8d8d8;
  opacity: 0.6;
`;

export const HeaderTab = () => {
  return (
    <View style={styles.container}>
      <Menu>Following</Menu>
      <Separator />
      <Menu bold="true">Explore</Menu>
    </View>
  );
};
