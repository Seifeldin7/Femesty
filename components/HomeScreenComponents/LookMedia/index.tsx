import React, { useEffect, useState } from "react"
import styled from "styled-components/native"
import { Text, View, TouchableOpacity } from "react-native"
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome'
import { faCircle } from '@fortawesome/free-solid-svg-icons'
import Product from '../../../entities/Product';
import { DotPopUp } from "../../SharedComponents/DotPopUp";
import { OrderItem } from "../../OrderScreenComponents"
import { styles } from './styles'
import { componentProps, screenHeight, screenWidth } from './types'
import { SecondaryColors } from "../../../styles/Colors"
import { FontStyles } from "../../../styles/Fonts"
import { Avatar } from "../../ThemeComponents/Avatar"
import { Ionicons } from "@expo/vector-icons"
import { responsiveScreenWidth } from "react-native-responsive-dimensions"

const Poster = styled.ImageBackground`
	height: 100%;
`

const LookMedia = ({
	poster,
	username,
	image,
	description,
	title,
	dots,
	index,
	firstMedia,
	pageNumSelected,
	navigation }: componentProps) => {
	const [clicked, setClicked] = useState(false);
	const [flag, setFlag] = useState(true);
	const [isModalVisible, setIsModalVisible] = useState(false);
	const [chosenProduct, setChosenProduct] = useState(new Product(1, '', '', 1, [], [], '', '', '', '', {}, new Date()));

	const media = <Poster source={{ uri: poster }} />;
	const wait = (timeout: any) => {
		return new Promise((resolve) => {
			setTimeout(resolve, timeout);
		});
	};
	useEffect(() => {
		if (index === pageNumSelected) {
			wait(1200).then(() => setFlag(false));
		}
	}, [pageNumSelected])

	const openModal = (product: Product) => {
		setChosenProduct(product);
		setIsModalVisible(true);
	}
	const isUpperScreen = (dotX: number, dotY: number) => {
		return (((dotY <= screenHeight / 2) || isLowerEdge(dotY)) && !isUpperEdge(dotY))
		//&& !isLeftOfScreen(dotX) && !isRightOfScreen(dotX) 
	}
	const isLowerScreen = (dotX: number, dotY: number) => {
		return (((dotY > screenHeight / 2) || isUpperEdge(dotY)) && !isLowerEdge(dotY))
		//&& !isLeftOfScreen(dotX) && !isRightOfScreen(dotX)
	}
	const isLeftOfScreen = (dotX: number) => {
		return (dotX <= screenWidth / 5)
	}
	const isRightOfScreen = (dotX: number) => {
		return (dotX >= (screenWidth - 100))
	}
	const isUpperEdge = (dotY: number) => {
		return (dotY <= 150)
	}
	const isLowerEdge = (dotY: number) => {
		return (dotY >= screenHeight - 150)
	}
	return (
		<TouchableOpacity activeOpacity={1.0} onPress={() => { setClicked(!clicked) }}>
			{media}
			<OrderItem
				product={chosenProduct}
				isVisible={isModalVisible}
				setModalVisible={setIsModalVisible}
				navigation={navigation} />

			{dots.map((dot: any, index: number) => {
				let pos = JSON.parse(dot.pos);
				let dotX: number = (pos.x) * screenWidth;
				let dotY: number = (pos.y) * screenHeight;
				let product: Product = dot.product;

				let productView =
					<View style={styles.product}>
						<Text style={{ ...FontStyles.SmallText }} >{product.name}</Text>
						<View style={{ flexDirection: 'row', justifyContent: 'space-between', marginTop: 5 }}>
							<Text style={{ ...FontStyles.SmallText }}>{product.seller.name}</Text>
							<Text style={{ marginLeft: 50, color: SecondaryColors.darkGrey, marginRight: 3 }}>{product.price} EGP</Text>
							<Ionicons
								name="ios-arrow-forward"
								size={responsiveScreenWidth(5)}
								color={SecondaryColors.darkGrey}
							/>
						</View>
					</View>;
				return (

					<View
						key={index}
						style={{ ...styles.dotStyle, marginTop: dotY, marginLeft: dotX - 100 }}>
						{((!flag && firstMedia) || !firstMedia) && !clicked ?
							<TouchableOpacity
								activeOpacity={1.0} onPress={() => { setClicked(true); }}
								style={{ alignItems: 'center' }}>
								<FontAwesomeIcon
									size={20}
									icon={faCircle} style={styles.dot} />
							</TouchableOpacity> : null}
						{(clicked && !flag) ?
							<DotPopUp
								visibility={1}
								arrowDirection={isUpperScreen(dotX, dotY) ? "up" : "down"}
								onCalloutPressed={() => { openModal(product); }}>
								{productView}
							</DotPopUp> : null}

					</View>
				)
			})}
			<View style={styles.textStyle}>
				<Text style={styles.modelTitle}>{title}</Text>
				<View style={{ flexDirection: 'row', alignItems: "center", paddingVertical: 5 }}>
					{(image == '') ?
						<Avatar imgPath={require("../../../assets/images/default_profile_pic.jpg")} /> :
						<Avatar imgPath={{ uri: image }} />
					}
					<Text style={styles.madeBy}>By: {username}</Text>
				</View>
				<Text style={styles.description}>{description}</Text>
			</View>
		</TouchableOpacity >
	)
}


export const MemoLookMedia = React.memo(LookMedia);
