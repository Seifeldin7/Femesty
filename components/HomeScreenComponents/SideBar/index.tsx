import React, { useState } from "react";
import { View, TouchableOpacity } from "react-native";
import { ShareLook } from "../../../utility/ShareLook";
import { FontAwesome } from "@expo/vector-icons";
import { styles } from "./styles";
import { componentProps } from "./types";
import { Icon } from "../../ThemeComponents/Icon";

export const Sidebar = ({
  look,
  addFavorite,
  deleteFavorite,
}: componentProps) => {
  const [heartColor, setHeartColor] = useState("white");
  return (
    <View style={styles.container}>
      <Icon
        outerStyle={{ marginBottom: 30, marginRight: 15 }}
        size="medium"
        iconPath={require("../../../assets/icons/small_icons/share.png")}
        onPress={() => ShareLook(look)} />

      <View>
        <TouchableOpacity
          onPress={() => {
            heartColor === "red"
              ? deleteFavorite(look.id)
                .then(() =>
                  setHeartColor((heartColor) =>
                    heartColor == "white" ? "red" : "white"
                  )
                )
                .catch()
              : addFavorite(look)
                .then(() =>
                  setHeartColor((heartColor) =>
                    heartColor == "white" ? "red" : "white"
                  )
                )
                .catch();
          }}
        >
          <View style={{ marginRight: 17 }}>
            {(heartColor == "white") ?
              <FontAwesome name="heart-o" size={28} color={heartColor} /> :
              <FontAwesome name="heart" size={28} color={heartColor} />
            }
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};
