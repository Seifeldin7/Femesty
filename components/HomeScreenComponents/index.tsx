export * from "./FilterButton"
export * from "./HeaderTab"
export * from "./HorizontalScroller"
export * from "./LookMedia"
export * from "./SideBar"