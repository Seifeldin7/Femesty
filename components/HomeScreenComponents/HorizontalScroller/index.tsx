import ViewPager from "@react-native-community/viewpager";
import React, { useState } from "react";
import { View } from "react-native";
import { Sidebar } from "../SideBar";
import { MemoLookMedia } from "../LookMedia";
import Dot from "../../../entities/Dot";

import { componentProps, screenHeight } from "./types";
import { styles } from "./styles";
import { Paginator } from "../../SharedComponents/Paginator";

const HorizontalScroller = ({
  look,
  addFavorite,
  deleteFavorite,
  navigation,
  index,
  pageNumSelected
}: componentProps) => {
  const [selectedPage, setSelectedpage] = useState(0);
  let pageNumber: number = 4;
  let media1: string = look.media1;
  let media2: string = look.media2;
  let media3: string = look.media3;
  let media4: string = look.media4;
  let dots: Dot[] = look.dots;
  let stylistName: string = look.stylist.name;
  let stylistImage: string = look.stylist.image;
  let name: string = look.name;
  let description: string = look.description;
  let media1Dots = [];
  let media2Dots = [];
  let media3Dots = [];
  let media4Dots = [];
  for (let dot of dots) {
    media1Dots.push({ pos: dot.media1Dot, product: dot.product });
    media2Dots.push({ pos: dot.media2Dot, product: dot.product });
    media3Dots.push({ pos: dot.media3Dot, product: dot.product });
    media4Dots.push({ pos: dot.media4Dot, product: dot.product });
  }

  return (
    <View style={{ flex: 1, flexDirection: "column", height: screenHeight }}>
      <Sidebar
        look={look}
        addFavorite={addFavorite}
        deleteFavorite={deleteFavorite}
      />
      <Paginator
        curPage={selectedPage}
        maxPages={pageNumber}
      />
      <ViewPager
        style={styles.fullscreen}
        orientation="horizontal"
        onPageSelected={(e) => setSelectedpage(e.nativeEvent.position)}
        initialPage={0}
      >
        <View>
          <MemoLookMedia
            poster={media1}
            username={stylistName}
            image={stylistImage}
            description={description}
            title={name}
            dots={media1Dots}
            firstMedia={true}
            navigation={navigation}
            index={index}
            pageNumSelected={pageNumSelected}
          />
        </View>
        <View>
          <MemoLookMedia
            poster={media2}
            username={stylistName}
            description={description}
            image={stylistImage}
            title={name}
            dots={media2Dots}
            firstMedia={false}
            navigation={navigation}
          />
        </View>
        <View>
          <MemoLookMedia
            poster={media3}
            username={stylistName}
            description={description}
            title={name}
            dots={media3Dots}
            image={stylistImage}
            firstMedia={false}
            navigation={navigation}
          />
        </View>
        <View>
          <MemoLookMedia
            poster={media4}
            username={stylistName}
            description={description}
            image={stylistImage}
            title={name}
            dots={media4Dots}
            firstMedia={false}
            navigation={navigation}
          />
        </View>
      </ViewPager>
    </View>
  );
};

export const MHorizontalScroller = React.memo(HorizontalScroller);
