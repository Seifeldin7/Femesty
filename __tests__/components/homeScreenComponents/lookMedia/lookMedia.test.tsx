import React from "react";
import { render } from "react-native-testing-library";
import { MemoLookMedia } from "../../../../components/HomeScreenComponents";
const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: () => mockDispatch
}));
jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: ''
}))
jest.mock('../../../../services/ApiClient', () => ({
    Constants: {
        manifest: {
            developer: {
                projectRoot: '/home/test/project',
            },
            logUrl: 'https://localhost:19001/logs',
            extra: {
                baseUrl: '192.168.1.1'
            }
        }
    },
}));
test('should render lookMedia correctly', () => {
    const tree = render(
        <MemoLookMedia
            image={"../../../../assets/images/01.jpg"}
            poster={"../../../../assets/images/01.jpg"}
            navigation= {jest.fn()}
            username={"pretty"}
            description={"Cool Look"}
            title={"Egyptian Look"}
            dots={[]}
            firstMedia={true}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
