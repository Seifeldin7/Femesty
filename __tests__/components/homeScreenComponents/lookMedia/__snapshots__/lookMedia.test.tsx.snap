// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render lookMedia correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "height": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "../../../../assets/images/01.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      testID="Popup"
    >
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        onRequestClose={[Function]}
        onSwipeComplete={[Function]}
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeDirection={
          Array [
            "down",
          ]
        }
        swipeThreshold={100}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
        />
        <View
          forwardedRef={[Function]}
          hideModalContentWhileAnimating={false}
          onModalHide={[Function]}
          onModalWillHide={[Function]}
          onModalWillShow={[Function]}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          onSwipeComplete={[Function]}
          pointerEvents="box-none"
          scrollHorizontal={false}
          scrollOffset={0}
          scrollOffsetMax={0}
          scrollTo={null}
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
              "left": 0,
              "margin": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          swipeDirection={
            Array [
              "down",
            ]
          }
          swipeThreshold={100}
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderTopLeftRadius": 20,
                  "borderTopRightRadius": 20,
                  "flex": 1,
                  "maxHeight": 800.4,
                  "width": 750,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#363636",
                      "borderRadius": 2.5,
                      "height": 5,
                      "marginTop": 6,
                      "width": 300,
                    }
                  }
                />
                <Text />
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flexDirection": "row",
                    "marginBottom": 8,
                    "width": 675,
                  }
                }
                testID="orderInfo"
              >
                <Image
                  source={
                    Object {
                      "uri": "",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 15,
                      "height": 80,
                      "width": 80,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "space-between",
                      "marginLeft": 40,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 28,
                      }
                    }
                  >
                    
                  </Text>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 22,
                        }
                      }
                    />
                    <Text
                      style={
                        Object {
                          "color": "grey",
                          "fontSize": 18,
                        }
                      }
                    >
                      1
                       EGP
                    </Text>
                  </View>
                </View>
              </View>
              <View
                testID="colorAndSizePickers"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 18,
                      "marginLeft": 37.5,
                    }
                  }
                >
                  Size
                </Text>
                <RCTScrollView
                  contentContainerStyle={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                  horizontal={true}
                >
                  <View />
                </RCTScrollView>
                <Text
                  style={
                    Object {
                      "fontSize": 18,
                      "marginLeft": 37.5,
                    }
                  }
                >
                  Color
                </Text>
                <RCTScrollView
                  contentContainerStyle={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                  horizontal={true}
                >
                  <View />
                </RCTScrollView>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "backgroundColor": "#FFA07A",
                      "borderRadius": 50,
                      "height": 50,
                      "justifyContent": "center",
                      "margin": 14,
                      "opacity": 1,
                      "padding": 15,
                      "width": 525,
                    }
                  }
                  testID="ButtonTouchableOpacity"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "white",
                        },
                        Object {
                          "fontFamily": "Lato_Regular",
                          "fontSize": 16,
                          "margin": 14,
                        },
                      ]
                    }
                  >
                    Order Item
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
    <View
      style={
        Object {
          "elevation": 1,
          "fontWeight": "800",
          "marginLeft": 15,
          "marginTop": 920,
          "position": "absolute",
          "width": "90%",
          "zIndex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.8)",
            "fontSize": 20,
            "fontWeight": "bold",
            "letterSpacing": -0.2,
          }
        }
      >
        Egyptian Look
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.8)",
            "fontSize": 14,
            "letterSpacing": -0.2,
            "marginTop": 6,
            "width": "80%",
          }
        }
      >
        By: 
        pretty
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.8)",
            "fontSize": 14,
            "letterSpacing": -0.2,
            "marginTop": 6,
            "width": "80%",
          }
        }
      >
        Cool Look
      </Text>
    </View>
  </View>
</View>
`;
